rdbi-recipes-java7/src/main/java/com/lithium/dbi/rdbi/recipes/cache/RedisCache.java

```diff
@@ -96,6 +96,22 @@ public Boolean run(Handle handle) {
         });
     }
 
+    public boolean acquireLockPatiently(KeyType key, int maxWaitMillis) {
+        final long startTime = System.currentTimeMillis();
+        boolean acquiredLock;
+        while (!(acquiredLock = acquireLock(key)) && System.currentTimeMillis() - startTime < maxWaitMillis) {
+            try {
+                TimeUnit.MILLISECONDS.sleep(250L);
+            } catch(InterruptedException ie) {
+                log.info("{}: interrupted acquiring lock", cacheName, ie);
```

Comment:
normally swallowing an interrupt is a bad idea, consider calling Thread.currentThread().interrupt() so that it can be handled elsewhere.